phoenix-core/src/main/java/org/apache/phoenix/execute/BaseQueryPlan.java

```diff
@@ -204,8 +204,18 @@ public final ResultIterator iterator(final List<? extends SQLCloseable> dependen
         } else {
             ScanUtil.setTimeRange(scan, context.getScanTimeRange());
         }
-        
-        ScanUtil.setTenantId(scan, connection.getTenantId() == null ? null : connection.getTenantId().getBytes());
+        byte[] tenantIdBytes;
```

Comment:
In both cases we are getting it from the connection. In the if {} block we are using the table to figure out the RowKeySchema so we know what to convert it into.

The else might be unnecessary, I leaned towards keeping it in because then it mimicked the old behaviour in the case where you were scanning a regular table on a tenant-specific connection.